### PR TITLE
pyrosimple: init at 2.7.0

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes
pyrosimple is a collection of tools for working with rTorrent and a Python3 rewrite of pyrocore.

Homepage: https://kannibalox.github.io/pyrosimple/

Its predecessor, pyrocore, has not been updated since 2017 and seems to be stuck with Python2.  See [its release page](https://github.com/pyroscope/pyrocore/releases).

Related PR: #60982

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ne9z** [comment]
@yorickvP As the author of pyrocore pull request, you might want to look into this.

**Comment by ne9z** [comment]
cc: @kannibalox

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (kannibalox) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:43]: Minor quibble: the license is GPLv3.

**Comment by ne9z** [comment]
Thanks.  I see that on the PyPi page, the license is listed as GPLv3 and
any later version.  Updated in this pull request.

**Comment by ne9z** [comment]
cc: @vamega

**Comment by vamega** [comment]
This looks good to me.
I had been wanting to do this, but was waiting for #215880 to get merged.

I'd also gone and packaged dynaconf in #214863, but since then @kannibalox removed the dependency on dynaconf.

You'll need to add yourself as a maintainer, it's why ofborg is failing.
That should be a separate commit. You can see the feedback I got on this in https://github.com/NixOS/nixpkgs/pull/214856

**Comment by ne9z** [comment]
Thanks for the helpful information!  I think it is now fixed.

**Comment by vamega** [comment]
This is minor, but you've added the commit listing you as a maintainer after the commit where you add pyrosimple.
That makes the commit you've added pyrosimple not build.

Could you rebase to re-order those commits?

**Comment by ne9z** [comment]
Thanks! Reordered.

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (yorickvP) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:15]:
```suggestion
    hash = "sha256-SMqzvTbWFHwnbMQ+6K0m1v+PybceQK5EHEuN8FB6SaU=";
```

File: pkgs/top-level/all-packages.nix
Comment (yorickvP) [inline pkgs/top-level/all-packages.nix:32901]:
```suggestion
  pyrosimple = python3Packages.callPackage ../applications/networking/p2p/pyrosimple { };
```

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (yorickvP) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:18]: move these into the function args instead of `python3Packages`, in conjunction with the all-packages change below

**Review by yorickvP — CHANGES_REQUESTED** [review]
Thanks for doing this! Some minor nitpicks, feel free to ignore.

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (vamega) [reply pkgs/applications/networking/p2p/pyrosimple/default.nix:45]: It'll work without `inotify` on other platforms.
It's listed [upstream as an optional dependency](https://github.com/kannibalox/pyrosimple/blob/6fee78e6342890c6503935a73f698abd4c6f5441/pyproject.toml#L43).

The functionality dependent on file system monitoring doesn't work, but otherwise it works fine.

I _think_ moving the dependencies to an argument, and having `inotify` omitted if the platform is not linux would be the right way to do this.

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (vamega) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:10]: Pretty sure this should be `apscheduler`
Fixing this should fix the ofborg failure.

**Comment by ne9z** [comment]
pyrosimple 2.7.1 requires a higher version of python-daemon than currently available in nixpkgs, which is 2.3.0.

```
       > ERROR: Could not find a version that satisfies the requirement python-daemon<4.0.0,>=3.0.1 (from pyrosimple) (from versions: none)
       > ERROR: No matching distribution found for python-daemon<4.0.0,>=3.0.1
       > 
       For full logs, run 'nix log /nix/store/pdynw24wah69c04wjdg7wf53avq4qs1j-python3.10-pyrosimple-2.7.1.drv'.
```

**Comment by vamega** [comment]
Looking at the [changelog](https://pagure.io/python-daemon/blob/main/f/ChangeLog) for `python-daemon`, it's only a bump for it's transitive dependency on setuptools, which is satisfied by the setuptools in nixpkgs.

You could you the `pythonRelaxDepsHook` to ignore that dependency bound. Maybe add a comment that it should be removed when next updating this package if `python-daemon` has been updated by then.

I had a few of these in my original derivation, but eventually submitted the CR's to upgrade the other packages so that wasn't necessary.

If you're sick of updating this, the current set of changes for 2.7.0 look good to me, can merge that and the update to 2.7. can come at another time.

Having a single PR is nice, since it reduces the number of PRs the nixpkgs committers have to review (i'm not a committer). Since the lead time can be long to get things reviewed, it's good to get it in earlier.

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (yorickvP) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:48]:
```suggestion
  ] ++ lib.optional withInotify inotify;
```

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (yorickvP) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:19]:
```suggestion
, inotify
, withInotify ? stdenv.isLinux
```

File: pkgs/applications/networking/p2p/pyrosimple/default.nix
Comment (yorickvP) [inline pkgs/applications/networking/p2p/pyrosimple/default.nix:3]:
```suggestion
```

**Review by yorickvP — CHANGES_REQUESTED** [review]
Thanks! `inotify` can also come from `python3Packages.callPackage`, I checked that that would use the right one :)

**Comment by ne9z** [comment]
I think next time when you have a suggestion, it would be easier, and
less error-prone, if you directly open a pull-request against my repo.

As a human, I'm likely to introduce unintended errors when applying
suggestions.  Especially when GitHub sends out so poorly formatted
emails.

**Review by yorickvP — APPROVED** [review]

**Comment by yorickvP** [comment]
Hi! Thanks for applying the changes. My idea was that you could auto-apply them using the github web-UI. I agree that the emails it sends out are quite bad.

**Pull request merged**
